This is a fun error

```
panic: division by zero

goroutine 4160 [running]:
math/big.nat.div({0xc001c58ba0?, 0x1?, 0x6?}, {0x0?, 0x0?, 0x0?}, {0xc001c58ba0?, 0x1?, 0x6?}, {0xc001d6a668, ...})
	math/big/natdiv.go:520 +0x32e
math/big.(*Int).QuoRem(0xc001c70b20, 0xc001c70b20, 0xc0016937e0, 0xc001693720)
	math/big/int.go:294 +0x7b
math/big.(*Int).Div(0xc001d6a668?, 0xc001c70b20?, 0xc001c70ae0?)
	math/big/int.go:305 +0x38
github.com/ethereum/go-ethereum/consensus/misc/eip1559.CalcBaseFee(0xc000a033b0, 0xc001dac000, 0x653a963f)
	github.com/ethereum/go-ethereum/consensus/misc/eip1559/eip1559.go:93 +0x547
github.com/ethereum/go-ethereum/miner.(*worker).prepareWork(0xc000412b40, 0xc000e7c790)
	github.com/ethereum/go-ethereum/miner/worker.go:987 +0x54d
github.com/ethereum/go-ethereum/miner.(*worker).generateWork(0xc000412b40, 0xc000e7c790)
	github.com/ethereum/go-ethereum/miner/worker.go:1066 +0x4f
github.com/ethereum/go-ethereum/miner.(*worker).mainLoop(0xc000412b40)
	github.com/ethereum/go-ethereum/miner/worker.go:556 +0x44c
created by github.com/ethereum/go-ethereum/miner.newWorker in goroutine 1
	github.com/ethereum/go-ethereum/miner/worker.go:294 +0x846
```

--- a/go.mod
+++ b/go.mod
@@ -8,7 +8,7 @@ require (
 	github.com/btcsuite/btcd/chaincfg/chainhash v1.0.1
 	github.com/decred/dcrd/dcrec/secp256k1/v4 v4.2.0
 	github.com/ethereum-optimism/go-ethereum-hdwallet v0.1.3
-	github.com/ethereum-optimism/superchain-registry/superchain v0.0.0-20231018202221-fdba3d104171
+	github.com/ethereum-optimism/superchain-registry/superchain v0.0.0-20231026175037-2cff0d130e74
 	github.com/ethereum/go-ethereum v1.13.1
 	github.com/fsnotify/fsnotify v1.7.0
 	github.com/go-chi/chi/v5 v5.0.10
@@ -209,7 +209,7 @@ require (
 	rsc.io/tmplfunc v0.0.3 // indirect
 )
 
-replace github.com/ethereum/go-ethereum v1.13.1 => github.com/ethereum-optimism/op-geth v1.101303.0-rc.2.0.20231024175019-29cd9a353f83
+replace github.com/ethereum/go-ethereum v1.13.1 => github.com/ethereum-optimism/op-geth v1.101303.0-rc.2.0.20231026180835-94fbbd04522e
 
 //replace github.com/ethereum-optimism/superchain-registry/superchain => ../superchain-registry/superchain
 //replace github.com/ethereum/go-ethereum v1.13.1 => ../go-ethereum

--- a/op-chain-ops/genesis/genesis.go
+++ b/op-chain-ops/genesis/genesis.go
@@ -65,9 +65,9 @@ func NewL2Genesis(config *DeployConfig, block *types.Block) (*core.Genesis, erro
 		CanyonTime:                    config.CanyonTime(block.Time()),
 		ShanghaiTime:                  config.CanyonTime(block.Time()),
 		Optimism: &params.OptimismConfig{
-			EIP1559Denominator:           eip1559Denom,
-			EIP1559Elasticity:            eip1559Elasticity,
-			EIP1559DenominatorPostCanyon: eip1559DenomCanyon,
+			EIP1559Denominator:       eip1559Denom,
+			EIP1559Elasticity:        eip1559Elasticity,
+			EIP1559DenominatorCanyon: eip1559DenomCanyon,
 		},
 	}
 

--- a/op-node/flags/flags.go
+++ b/op-node/flags/flags.go
@@ -256,7 +256,7 @@ var (
 	CanyonOverrideFlag = &cli.Uint64Flag{
 		Name:   "override.canyon",
 		Usage:  "Manually specify the Canyon fork timestamp, overriding the bundled setting",
-		Hidden: true,
+		Hidden: false,
 	}
 )
 

--- a/op-node/rollup/superchain.go
+++ b/op-node/rollup/superchain.go
@@ -12,7 +12,7 @@ import (
 	"github.com/ethereum-optimism/superchain-registry/superchain"
 )
 
-var OPStackSupport = params.ProtocolVersionV0{Build: [8]byte{}, Major: 3, Minor: 1, Patch: 0, PreRelease: 1}.Encode()
+var OPStackSupport = params.ProtocolVersionV0{Build: [8]byte{}, Major: 4, Minor: 0, Patch: 0, PreRelease: 1}.Encode()
 
 const (
 	opMainnet   = 10
@@ -98,6 +98,7 @@ func LoadOPStackRollupConfig(chainID uint64) (*Config, error) {
 		L1ChainID:              new(big.Int).SetUint64(superChain.Config.L1.ChainID),
 		L2ChainID:              new(big.Int).SetUint64(chConfig.ChainID),
 		RegolithTime:           &regolithTime,
+		CanyonTime:             superChain.Config.CanyonTime,
 		BatchInboxAddress:      common.Address(chConfig.BatchInboxAddr),
 		DepositContractAddress: depositContractAddress,
 		L1SystemConfigAddress:  common.Address(chConfig.SystemConfigAddr),

